pkg/daemon/ceph/osd/agent.go

```diff
@@ -571,6 +571,11 @@ func waitForRebalance(context *clusterd.Context, clusterName string, osdID int,
 					osdID, initialUsage, currUsage)
 			}
 
+			// when the initial OSD PG count or the used KB is zero there is nothing to do here
+			if init.Pgs == "0" || init.UsedKB == "0" {
```

Comment:
when i originally wrote this function, I don't think I realized that `json.Number` type is just a string.  The comparisons are not valid in this function, see this quick example: https://play.golang.org/p/PQ-dmFp2PdN

I think we should convert these to integers first with https://golang.org/pkg/encoding/json/#Number.Int64 and then make any comparisons.  Can we do that now please? :anguished: